web-common/src/features/canvas/RowBasedCanvas.svelte

```diff
@@ -75,23 +71,27 @@
   let dragTimeout: ReturnType<typeof setTimeout> | null = null;
   let dragItemPosition = { top: 0, left: 0 };
   let dragItemDimensions = { width: 0, height: 0 };
-  let spec = $canvasSpec ?? {
+  let spec = canvasData?.canvas?.canvas?.spec ?? {
     rows: [],
     filtersEnabled: false,
     maxWidth: DEFAULT_DASHBOARD_WIDTH,
   };
   let openSidebarAfterSelection = false;
 
   $: ({ instanceId } = $runtime);
+  $: metricsViews = Object.values(canvasData?.referencedMetricsViews ?? {});
```

Comment:
We should use the `useCanvas` selector upstream so these sort of references can be abstracted.